src/databricks/labs/ucx/workspace_access/groups.py

```diff
@@ -200,18 +202,40 @@ def _account_groups_in_account(self) -> dict[str, str]:
             by_name[g.display_name] = g.id
         return by_name
 
+    def _is_group_out_of_scope(self, group: iam.Group, resource_type: str) -> bool:
+        if group.display_name in self._SYSTEM_GROUPS:
+            return True
+        if group.meta.resource_type != resource_type:
+            return True
+        return False
+
     def _list_workspace_groups(self, resource_type: str, scim_attributes: str) -> list[iam.Group]:
         results = []
         logger.info(f"Listing workspace groups (resource_type={resource_type}) with {scim_attributes}...")
-        for g in self._ws.groups.list(attributes=scim_attributes):
-            if g.display_name in self._SYSTEM_GROUPS:
-                continue
-            if g.meta.resource_type != resource_type:
-                continue
-            results.append(g)
+        # these attributes can get too large causing the api to timeout
+        # so we're fetching groups without these attributes first
+        # and then calling get on each of them to fetch all attributes
+        attributes = scim_attributes.split(",")
+        if "members" in attributes:
+            attributes.remove("members")
+            for g in self._ws.groups.list(attributes="id,displayName,meta"):
```

Comment:
```suggestion
            for g in self._ws.groups.list(attributes=",".join(attributes)):
```
otherwise it's a bug